### ISSUE TITLE
1. Übungsblatt: UML-Klassen- und -Objektdiagramme Aufgabe 5

### ISSUE THREAD

**Issue by hexahepta** [sweIhm-ws2018-19/skillproject-fr_13#2]

**Comment by HoferAnton** [comment]
Änderung an Objektnamen -> Objektnamen starten mit Kleinbuchstaben und sind einzigartig.

**Comment by OneToughMonkey** [comment]
Habe hier nochmal kleine Anpassungen vorgenommen, bitte Commit-Message beachten.

https://github.com/sweIhm-ws2018-19/skillproject-fr_13/commit/4001b5866736966bb4d1e877b051d9287892a225

**Comment by hexahepta** [comment]
@OneToughMonkey Danke damit hast du natürlich recht.
@HoferAnton bitte nochmal hier bestätigen wenn damit alles passt, dann werde ich das Issue resolven.

**Comment by HoferAnton** [comment]
@hexahepta Werden die auf den Assoziationen angedeuteten Attribute in das Objektdiagramm übrenommen?

**Comment by hexahepta** [comment]
@HoferAnton Ja, ich verstehe die Links im Objektdiagramm als Instanzen der Relationen aus dem Klassendiagramm.

**Comment by HoferAnton** [comment]
@hexahepta Genau mit den Links selbst stimme ich überein, die Frage ist eher wie sich der Rollenname aus dem Klassendiagramm  auf das Objektdiagramm auswirkt. So wird im Buch, bei den Beispielen für Objektdiagramme Abb. 4.5-1, S. 290, die Rolle auch auf den Link geschrieben. In unserem Fall müsste dann doch bei der Adresse auf den Links noch postAdresse bzw. domizilAdresse stehen oder habe ich da etwas missverstanden?

**Comment by OneToughMonkey** [comment]
Bei vielen Objektdiagrammen die ich gesehen habe, z.B. auch denen auf den Foliensätzen, werden die Rollennamen auch weggelassen. Der Detaillierungsgrad eines Diagramms kann ja je nach Kontext variieren und bildet immer einen Trade-Off zwischen Vollständigkeit und Übersichtlichkeit.

Generell würd ich mir, wenn ein Objektdiagramm in direktem Zusammenhang mit seinem Klassendiagramm betrachtet wird, die Rollen dort schon ersichtlich und die Rollenbeziehungen der Klassen zueinander eindeutig sind, eher zweimal überlegen, ob es sinnvoll ist die Information im Objektdiagramm nochmal zu wiederholen.

Nachdem der Kontext bei dieser Aufgabe aber nicht ganz klar definiert ist stimme ich dafür, dass wir die Rollenbezeichnungen noch in das Diagramm aufnehmen.

**Comment by hexahepta** [comment]
@OneToughMonkey hast du da schon eine konkrette Vorstellung das Umzusetzten? Dann würde ich das dir überlassen, sonnst mach ich das noch.

**Comment by OneToughMonkey** [comment]
@hexahepta done
https://github.com/sweIhm-ws2018-19/skillproject-fr_13/commit/83fb45c2e1caf82d3f3e9c4acddae2923356e8e2

**Comment by hexahepta** [comment]
@OneToughMonkey nice danke, sehe ich mir gleich mal an.
@HoferAnton bitte gib auch du als offizieller Reviewer noch dein Urteil ab.

**Comment by hexahepta** [comment]
Sieht für mich ganz in Ordnung aus. Sobald @HoferAnton auch sein ok gibt, resolve ich das Issue. Dann müssen wir nur noch den Branch zu master mergen und die Lösungen im Wiki verlinken.

**Comment by HoferAnton** [comment]
@hexahepta und @OneToughMonkey Ja schaut gut aus denke kann man so lassen.
